add_warranty/wizard/add_warranty_wizard.py

```diff
@@ -0,0 +1,60 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError, ValidationError
+
+class AddWarrantyLinesWizard(models.TransientModel):
+    _name = "add.warranty.lines.wizard"
+    _description = "Add Warranty Lines Wizard"
+
+    wizard_id = fields.Many2one('add.warranty.wizard', string="Wizard Reference", ondelete="cascade", required=True)
+    product_id = fields.Many2one(comodel_name="product.product", string="Product")
+    sale_order_line_id = fields.Many2one(comodel_name="sale.order.line", string="Sale order lines")
+    year_id = fields.Many2one(comodel_name="warranty.config", string="Year")
+    end_date = fields.Date(readonly=True, string="End Date", compute="_compute_end_date")
+
+    @api.depends('year_id')
+    def _compute_end_date(self):
+        for record in self:
+            if record.year_id:
+                record.end_date = fields.Date.today() + relativedelta(days=365 * record.year_id.period)
+            else:
+                record.end_date = False 
```

Comment:
```suggestion
    @api.depends('year_id')
    def _compute_end_date(self):
        for record in self:
            record.end_date = record.year_id and fields.Date.today() + relativedelta(days=365 * record.year_id.period) or False
```
is there need of adding else condition here? Could you please try this.